src/compiler/checker.ts

```diff
@@ -28127,6 +28134,11 @@ namespace ts {
             return suggestion;
         }
 
+        function getSuggestedTypeForNonexistentStringLiteralType(source: StringLiteralType, target: UnionType): StringLiteralType | undefined {
+            const candidates = target.types.filter((type): type is StringLiteralType => !!(type.flags & TypeFlags.StringLiteral));
+            return getSpellingSuggestion(source.value, candidates, (type) => type.value);

```

Comment:
very minor: no parens around arrow parameter
```suggestion
            return getSpellingSuggestion(source.value, candidates, type => type.value);
```